[DDW-430] Smart tooltips

DeeJayElly opened the pull request:
This PR implements Smart Tooltips component across whole Daedalus application

## Screenshots

<img width="1147" alt="Screen Shot 2020-11-21 at 11 33 54 AM" src="https://user-images.githubusercontent.com/15862062/99876339-7b914800-2bf6-11eb-8a89-3a3b620f7e59.png">

<img width="1145" alt="Screen Shot 2020-11-21 at 11 34 10 AM" src="https://user-images.githubusercontent.com/15862062/99876347-8055fc00-2bf6-11eb-9799-4aeb4c1172fc.png">

<img width="1146" alt="Screen Shot 2020-11-21 at 11 35 19 AM" src="https://user-images.githubusercontent.com/15862062/99876348-80ee9280-2bf6-11eb-8bbc-41a9439e1a64.png">

<img width="1145" alt="Screen Shot 2020-11-21 at 11 35 33 AM" src="https://user-images.githubusercontent.com/15862062/99876349-81872900-2bf6-11eb-9612-b835333ac8d9.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 35 46 AM" src="https://user-images.githubusercontent.com/15862062/99876350-821fbf80-2bf6-11eb-970a-bab6340bab81.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 35 53 AM" src="https://user-images.githubusercontent.com/15862062/99876351-82b85600-2bf6-11eb-8643-b1d8a9a3ea29.png">

<img width="1148" alt="Screen Shot 2020-11-21 at 11 36 05 AM" src="https://user-images.githubusercontent.com/15862062/99876352-82b85600-2bf6-11eb-9b09-4e3a78eea553.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 36 17 AM" src="https://user-images.githubusercontent.com/15862062/99876353-8350ec80-2bf6-11eb-9c56-58d8ffc40209.png">

<img width="1146" alt="Screen Shot 2020-11-21 at 11 36 30 AM" src="https://user-images.githubusercontent.com/15862062/99876354-83e98300-2bf6-11eb-9076-9124b84f561e.png">

<img width="1144" alt="Screen Shot 2020-11-21 at 11 37 10 AM" src="https://user-images.githubusercontent.com/15862062/99876356-851ab000-2bf6-11eb-99cf-011145f3fb38.png">

<img width="1146" alt="Screen Shot 2020-11-21 at 11 37 30 AM" src="https://user-images.githubusercontent.com/15862062/99876357-85b34680-2bf6-11eb-9f47-830891b933e7.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 38 03 AM" src="https://user-images.githubusercontent.com/15862062/99876358-85b34680-2bf6-11eb-8ffc-9c0d594783f8.png">

<img width="1146" alt="Screen Shot 2020-11-21 at 11 40 44 AM" src="https://user-images.githubusercontent.com/15862062/99876360-864bdd00-2bf6-11eb-8043-0b0fa26ae2d5.png">

<img width="1148" alt="Screen Shot 2020-11-21 at 11 40 56 AM" src="https://user-images.githubusercontent.com/15862062/99876361-86e47380-2bf6-11eb-9e2e-fe651530d53d.png">

<img width="1150" alt="Screen Shot 2020-11-21 at 11 41 10 AM" src="https://user-images.githubusercontent.com/15862062/99876362-877d0a00-2bf6-11eb-95d3-c686d7f13e7d.png">

<img width="1145" alt="Screen Shot 2020-11-21 at 11 41 23 AM" src="https://user-images.githubusercontent.com/15862062/99876363-877d0a00-2bf6-11eb-8ca8-0bc4c02c4bf1.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 41 42 AM" src="https://user-images.githubusercontent.com/15862062/99876364-8815a080-2bf6-11eb-90ab-d5a77f7c2b9f.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 11 41 50 AM" src="https://user-images.githubusercontent.com/15862062/99876365-88ae3700-2bf6-11eb-9150-6a13b244357a.png">

<img width="1149" alt="Screen Shot 2020-11-21 at 11 44 29 AM" src="https://user-images.githubusercontent.com/15862062/99876366-88ae3700-2bf6-11eb-87ee-057d690f8be2.png">

<img width="1144" alt="Screen Shot 2020-11-21 at 11 44 36 AM" src="https://user-images.githubusercontent.com/15862062/99876367-8946cd80-2bf6-11eb-92d1-d5e60a9e22fe.png">

<img width="1143" alt="Screen Shot 2020-11-21 at 11 44 44 AM" src="https://user-images.githubusercontent.com/15862062/99876368-89df6400-2bf6-11eb-9586-0b278e598271.png">

<img width="1151" alt="Screen Shot 2020-11-21 at 11 44 58 AM" src="https://user-images.githubusercontent.com/15862062/99876369-8a77fa80-2bf6-11eb-9ce8-3f07f9284c82.png">

<img width="1147" alt="Screen Shot 2020-11-21 at 12 34 35 PM" src="https://user-images.githubusercontent.com/15862062/99876371-8a77fa80-2bf6-11eb-9263-7d546cdaf533.png">

<img width="1143" alt="Screen Shot 2020-11-21 at 12 35 04 PM" src="https://user-images.githubusercontent.com/15862062/99876372-8b109100-2bf6-11eb-9dcb-6084dc777c66.png">

<img width="1457" alt="Screen Shot 2020-11-21 at 12 42 57 PM" src="https://user-images.githubusercontent.com/15862062/99876888-54d51080-2bfa-11eb-901e-859036040a0a.png">

<img width="1459" alt="Screen Shot 2020-11-21 at 12 49 40 PM" src="https://user-images.githubusercontent.com/15862062/99876890-59012e00-2bfa-11eb-959f-bc3ea774dfda.png">

<img width="1447" alt="Screen Shot 2020-11-21 at 12 53 56 PM" src="https://user-images.githubusercontent.com/15862062/99876892-5acaf180-2bfa-11eb-95c5-a7d7b2be95a1.png">

<img width="1456" alt="Screen Shot 2020-11-21 at 12 54 09 PM" src="https://user-images.githubusercontent.com/15862062/99876893-5acaf180-2bfa-11eb-9314-f58f2e378a7a.png">

<img width="1455" alt="Screen Shot 2020-11-21 at 12 54 48 PM" src="https://user-images.githubusercontent.com/15862062/99876895-5b638800-2bfa-11eb-9552-fc687f2e7e5b.png">

<img width="1461" alt="Screen Shot 2020-11-21 at 12 55 02 PM" src="https://user-images.githubusercontent.com/15862062/99876896-5c94b500-2bfa-11eb-963b-6d67e654389e.png">

<img width="1459" alt="Screen Shot 2020-11-21 at 12 55 14 PM" src="https://user-images.githubusercontent.com/15862062/99876899-5d2d4b80-2bfa-11eb-9af1-cb85b722e8ed.png">

<img width="1148" alt="Screen Shot 2020-11-21 at 1 08 12 PM" src="https://user-images.githubusercontent.com/15862062/99876944-b72e1100-2bfa-11eb-8f0c-cf8e7e0e56d3.png">

<img width="1149" alt="Screen Shot 2020-11-21 at 1 07 54 PM" src="https://user-images.githubusercontent.com/15862062/99876946-bb5a2e80-2bfa-11eb-8c6b-6fb87bb60b8e.png">

---

## Testing Checklist

- [Slack QA thread]()
- Check the following tooltips:
- [x] Connecting/Loading Screen - All Status Icons
- [x] Syncing status
- [x] Create Wallet Wizard - Enter Password Input
- [x] Restore Wallet Wizard -  Enter Password Input
- [x] Byron Wallet Settings - Paper Wallet Disclaimer
- [x] Wallet Settings - Change Password - Current Password Input
- [x] Daedalus Diagnostics - Daedalus State Directory
- [x] Daedalus Diagnostics - CPU
- [x] Stake Pool Details - All listed items
- [x] Stake Pool Details - Stake Pool ID
- [x] Delegation center - Delegated wallet
- [x] Stake Pools View - Saturation bar values


## Testing Summary

For build 15243:

**New**:
1 - Tooltips with long text are too wide. Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970).

**Rejected:**
~1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) - Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)~

**Fixed**
1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
2 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).`
3 - The pool id component moves for an instant after copying the value. [Check this recording](https://drive.google.com/file/d/1lMUy-2rMkdrbIgmYP7Pwd3oQPPq5sldC/view?usp=sharing). - [Reported here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-729734017)
4 -When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password.  [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) - Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
5 - The issue reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) about the ? icon and the tooltip (New # 2) still reproduces when restoring a wallet. Check [screenshot](https://drive.google.com/file/d/1SQTTCaavLdhy_FyVaKzgZPk1UkL53FqB/view?usp=sharing).

---

## Review Checklist

### Basics

- [x] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [x] In English
  - [x] In Japanese
- [x] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [x] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn flow:test`)
- [x] There are no *lint* errors or warnings (`yarn lint`)
- [x] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

gnpf commented:
@DeeJayElly here a list of some issues on build 15182:

1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).

2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)

DeeJayElly commented:
> @DeeJayElly here a list of some issues on build 15182:
> 
> 1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).
> 
> 2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)
> 
> I added to the description of this PR a list of the tooltips to check @mioriohk @ManusMcCole

2. Issue is fixed.

<img width="1148" alt="Screen Shot 2020-11-13 at 10 32 38 PM" src="https://user-images.githubusercontent.com/15862062/99123379-58680680-2600-11eb-9d9d-c84a914c303a.png">

DeeJayElly commented:
> @DeeJayElly here a list of some issues on build 15182:
> 
> 1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).
> 
> 2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)
> 
> I added to the description of this PR a list of the tooltips to check @mioriohk @ManusMcCole

1. Issue fixed

nikolaglumac commented:
@gnpf please re-check 🙏

nikolaglumac commented:
@DominikGuzei please review this one 🙏

mioriohk commented:
Hi @DeeJayElly, this is the summary for build 15186

For fixed issues, I have moved them to the top of this PR.

**New issue**
1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing)
 
**Suggestion**
1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? Also occurred during Create Wallet, Restore Wallet and Change password [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing)

DeeJayElly commented:
> Hi @DeeJayElly, this is the summary for build 15186
> 
> For fixed issues, I have moved them to the top of this PR.
> 
> **New issue**
> 1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing)
> 
> **Suggestion**
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing)

@DominikGuzei do we have tabIndex={-1} implementation inside PopOver component? I believe that needs to be added.

DominikGuzei commented on source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js:
```diff
@@ -113,16 +112,15 @@ export class StakePoolsSearch extends Component<Props> {
+                <PopOver
+                  content="Clear"
+                  contentClassName={styles.clearSearch}
+                  placement={!isClearTooltipOpeningDownward ? 'top': 'bottom'}
```
small thing but here it would be clearer to remove the negation: `isClearTooltipOpeningDownward ? 'bottom': 'top`

DeeJayElly replied:
Fixed

DominikGuzei commented on source/renderer/app/components/staking/widgets/TooltipPool-copyId-tooltip.scss:
```diff
@@ -1,19 +1,17 @@
+  .tippy-arrow {
+    &:before,
+    &:after {
+      margin-left: 45px;
```
Can you provide some context why this was necessary? I'm just asking because there is also a Tippy prop for specifying the offsets in JS: https://atomiks.github.io/tippyjs/v6/all-props/#offset

DeeJayElly replied:
Removed, its obsolete now.

DominikGuzei commented on source/renderer/app/components/wallet/WalletCreateDialog.scss:
```diff
@@ -81,9 +73,16 @@
+        position: relative;
+        top: 5px;
+        width: 12px;
+      }
```
☝️ is this css necessary for the tooltip? it would be great to get rid of any unnecessary css overrides as they are hard to understand

DeeJayElly replied:
Fixed

DominikGuzei commented on source/renderer/app/components/wallet/WalletRestoreDialog.scss:
```diff
@@ -108,9 +100,16 @@
+          left: -147px;
+          position: relative;
+          top: 2px;
+          width: 12px;
```
Same here, you can check https://atomiks.github.io/tippyjs/v6/all-props for available props that might be used instead

DeeJayElly replied:
Fixed

DominikGuzei commented on source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.scss:
```diff
@@ -39,10 +39,9 @@
+        left: 142px;
         position: absolute;
         top: 54px;
         width: 12px;
```
Same here, position of the tooltip should be done by the PopOver (and adjusted via Tippy props if necessary) - of course if there is a special case im not yet familiar with, please show me :)

DeeJayElly replied:
Fixed

DominikGuzei commented on source/renderer/app/components/wallet/wallet-restore/ConfigurationDialog.scss:
```diff
@@ -32,10 +32,10 @@
+      position: relative;
       top: 1px;
       width: 12px;
     }
```
Same here, would prefer to ged rid of any css overrides

DeeJayElly replied:
Fixed

DominikGuzei reviewed (changes requested):
Great work Alex 🎉 the code looks good apart from some questions regarding CSS overrides for some of the tooltips - i left you a bunch of comments. I haven't debugged the specifics but it would be great if we could ged rid of all css overrides if possible. Let's discuss on Slack!

gnpf commented:
@DeeJayElly The issues I reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727025104) are fixed (I'm updating the testing summary for this PR on the description).
I also added new elements on the testing checklist @mioriohk @ManusMcCole 

Here's a summary of new and pending issues for build 15200:

**New**
1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
2 - The `?` Icon and the tooltip are overlapping the text in Japanese. [See screenshot](https://drive.google.com/file/d/1HdE-kBe4HZVfetH7BBrP4BigFrIfkLx4/view?usp=sharing).
3 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).

**Pending**
1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password.  [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
 
**Suggestion**
1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)

DeeJayElly commented:
> @DeeJayElly The issues I reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727025104) are fixed (I'm updating the testing summary for this PR on the description).
> I also added new elements on the testing checklist @mioriohk @ManusMcCole
> 
> Here's a summary of new and pending issues for build 15200:
> 
> **New**
> 1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
> 2 - The `?` Icon and the tooltip are overlapping the text in Japanese. [See screenshot](https://drive.google.com/file/d/1HdE-kBe4HZVfetH7BBrP4BigFrIfkLx4/view?usp=sharing).
> 3 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).
> 
> **Pending**
> 1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
> 
> **Suggestion**
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)

Hi @gnpf @mioriohk @ManusMcCole all the requested changes and fixes have been applied. Pls retest it.

DominikGuzei reviewed (approved):
Amazing cleanup, thanks! 🙏 code looks good now 🎉

gnpf commented:
@DeeJayElly for build 15215:

**New**:
1 - There pool id component moves for an instant after copying the value. [Check this recording](https://drive.google.com/file/d/1lMUy-2rMkdrbIgmYP7Pwd3oQPPq5sldC/view?usp=sharing).

**Reincident**:
1 - The issue reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) about the ? icon and the tooltip (New # 2) still reproduces when restoring a wallet. Check [screenshot](https://drive.google.com/file/d/1SQTTCaavLdhy_FyVaKzgZPk1UkL53FqB/view?usp=sharing).

**Pending:**
1 - From [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) `pending issue # 1 `and `suggestion # 1`

**Fixed**
1 - 1 - From [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) `new issue # 1` and `new issue # 3`

DominikGuzei reviewed (changes requested):
I just realized that this PR is using an older version of react-polymorph (that's also why the tabbing issue exists, which i already fixed in the latest version).

So please update to 0.9.7-rc.5 and double-check the width of all tooltips you upgraded, they all need a `maxWidth` (pixel as number) prop otherwise they will stretch over the full width by default

DeeJayElly commented:
> I just realized that this PR is using an older version of react-polymorph (that's also why the tabbing issue exists, which i already fixed in the latest version).
> 
> So please update to 0.9.7-rc.5 and double-check the width of all tooltips you upgraded, they all need a `maxWidth` (pixel as number) prop otherwise they will stretch over the full width by default

Updated @DominikGuzei

DominikGuzei commented:
@DeeJayElly you updated to latest react-polymorph but forgot to add the `maxWidth` props to all tooltips with long content like here:
![grafik](https://user-images.githubusercontent.com/172414/99691816-e5d8aa00-2a89-11eb-92f8-2850edcb1561.png)

gnpf commented:
@DeeJayElly For build **15243** I updated the summary of this PR.
Besides [the issue with the widht of the tooltips](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970) mentioned by @DominikGuzei there are now new incidences.

Would this suggestion reported by @mioriohk be addressed on this PR? Maybe this needs to be discussed @nikolaglumac ?

> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? See here

It was reported on [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587).

nikolaglumac commented:
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? See here

No, this should remain as is. This is a different case and it has been implemented according to the original design.

nikolaglumac commented:
The only issue which remains here is the one @DominikGuzei reported in https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970

cc @DeeJayElly @gnpf

DeeJayElly commented:
> @DeeJayElly you updated to latest react-polymorph but forgot to add the `maxWidth` props to all tooltips with long content like here:
> ![grafik](https://user-images.githubusercontent.com/172414/99691816-e5d8aa00-2a89-11eb-92f8-2850edcb1561.png)

Added

DeeJayElly commented:
> The only issue which remains here is the one @DominikGuzei reported in [#2243 (comment)](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970)
> 
> cc @DeeJayElly @gnpf

Fixed @gnpf @nikolaglumac @DominikGuzei

gnpf commented:
@DeeJayElly All tooltips look good, except the [Daedalus Paper Wallet Disclaimer](https://drive.google.com/file/d/19-9nEL6MmP0AKC2W1nWSTFy9DhUBDkik/view?usp=sharing) (Text is out of the component)

DeeJayElly commented:
> @DeeJayElly All tooltips look good, except the [Daedalus Paper Wallet Disclaimer](https://drive.google.com/file/d/19-9nEL6MmP0AKC2W1nWSTFy9DhUBDkik/view?usp=sharing) (Text is out of the component)

Check it now @gnpf , it should be ok now.

nikolaglumac commented:
@gnpf please test again 🙏

gnpf reviewed (approved):
All tooltips look good, great job @DeeJayElly

nikolaglumac reviewed (changes requested):
@DeeJayElly @a-rukin @gnpf how can this one be approved - don't you see the line-height of the tooltip content? 😢 

![Screenshot 2020-11-20 at 17 40 38](https://user-images.githubusercontent.com/376611/99825814-d32ca680-2b57-11eb-83a4-8756f3eb0860.png)

a-rukin commented:
https://zpl.io/2p5dz9l - here you can find a proper line height for the tooltip (15px)

a-rukin commented:
Font size - 14px 
Line height - 19px

nikolaglumac commented:
@DeeJayElly please check **every single tooltip** and post screenshots of the fixed state for each of them both in English and Japanese. The font size and line-heights should be as @a-rukin stated. Thanks!

DeeJayElly commented:
> Font size - 14px
> Line height - 19px

Added

nikolaglumac reviewed (changes requested):
@DeeJayElly please add screenshots of every single tooltip affected by this PR in both English and Japanese.

DeeJayElly commented:
> @DeeJayElly please add screenshots of every single tooltip affected by this PR in both English and Japanese.

Added

mioriohk reviewed (approved):
Looks good to me 👍

a-rukin commented:
It seems to me that this distance is bigger (+ arrow shifted to the left)
<img width="373" alt="image" src="https://user-images.githubusercontent.com/2539345/99948333-60574180-2d8a-11eb-94f6-1de8c38faa5c.png">

DeeJayElly commented:
> It seems to me that this distance is bigger (+ arrow shifted to the left)
> <img alt="image" width="373" src="https://user-images.githubusercontent.com/2539345/99948333-60574180-2d8a-11eb-94f6-1de8c38faa5c.png">

Fixed @a-rukin

a-rukin commented:
<img width="707" alt="image" src="https://user-images.githubusercontent.com/2539345/99967534-13369800-2da9-11eb-955c-21f88ada270a.png">
it is still shifted to the left

a-rukin reviewed (approved):
ok

DominikGuzei reviewed (approved):
Code is looking good now 👍

DominikGuzei commented:
Great work @DeeJayElly 🎉 let's wait for final approval by @nikolaglumac

nikolaglumac commented on source/renderer/app/themes/overrides/PopOverOverrides.scss:
```diff
@@ -1,2 +1,5 @@
 .root {
+  :global .tippy-content {
```
@DominikGuzei isn't there a better way to target the content and set up the line-height?

DominikGuzei replied:
@nikolaglumac good point, we could probably 1. add a sane default in react-polymorph and 2. make it configurable via theme variable 👍 I also saw that most of the `maxWidth` values are 350, so I will also add that as default

nikolaglumac replied:
Will you take care of that?
